cocos/scripting/js-bindings/jswrapper/jsc/ScriptEngine.hpp

```diff
@@ -225,18 +232,6 @@ namespace se {
         void _retainScriptObject(void* owner, void* target);
         void _releaseScriptObject(void* owner, void* target);
 
-        enum class NodeEventType
-        {
-            ENTER,
-            EXIT,
-            ENTER_TRANSITION_DID_FINISH,
-            EXIT_TRANSITION_DID_START,
-            CLEANUP
-        };
-        bool _onReceiveNodeEvent(void* node, NodeEventType type);
-        using NodeEventListener = bool(*)(void*, NodeEventType);
-        bool _setNodeEventListener(NodeEventListener listener);
```

Comment:
jswrapper should be clean which means it should not depend on any concepts in Cocos2d-x. 
So, node event callback should be moved.